client/components/common/tce-core/Discussion/index.vue

```diff
@@ -54,17 +57,27 @@ export default {
   inheritAttrs: true,
   props: {
     comments: { type: Array, default: () => [] },
-    user: { type: Object, required: true },
+    unseenComments: { type: Array, default: () => [] },
+    commentsShownLimit: { type: Number, default: 5 },
+    scrollTarget: { type: String, default: 'discussion' },
     showHeading: { type: Boolean, default: false },
     showNotifications: { type: Boolean, default: false },
-    commentsShownLimit: { type: Number, default: 5 },
-    scrollTarget: { type: String, default: 'discussion' }
+    isActivityThread: { type: Boolean, default: false },
+    user: { type: Object, required: true }
   },
   data: () => ({ showAll: false, comment: initCommentInput() }),
   computed: {
-    thread: vm => orderBy(vm.comments, ['createdAt'], ['asc']),
+    thread() {
+      const { comments, unseenComments } = this;
+      const processedThread = comments.map(comment => {
+        const unseen = unseenComments.find(it => it.id === comment.id);
+        return { ...comment, unseen: !!unseen };
```

Comment:
No need to pass unseenArray if on the processed array you are passing it as attr (you can just pass items and rely on this for other usages aswell, much cleaner approach; you can also get rid of passing count and last seen dates later on)